### PR TITLE
Feat: Add Parametric Widths and Nat Literals to Automata Tactic

### PR DESCRIPTION
**Pull request by AtticusKuhn**
Added the following 2 features to the Automata Tactic:
1. Handling equalities over some parametric width w
2. Handling expression with nat literals

### CODE REVIEW

File: SSA/Experimental/Bits/Fast/Tactic.lean [hunk 1]
```diff
@@ -155,7 +207,12 @@ macro "bv_automata" : tactic =>
   native_decide
   ))
 
-def test1 (x y : BitVec 300) : (x ||| y) - (x ^^^ y) = x &&& y := by
+
+
+
+def test0 {w : Nat} (x y : BitVec (w + 1)) : x + 0 = x := by
+  bv_automata
+def test1 {w : Nat} (x y : BitVec (w + 1)) : (x ||| y) - (x ^^^ y) = x &&& y := by
```
Comment (tobiasgrosser) [inline SSA/Experimental/Bits/Fast/Tactic.lean]:
```suggestion
  bv_automata
def test1 {w : Nat} (x y : BitVec (w + 1)) : (x ||| y) - (x ^^^ y) = x &&& y := by

```

File: SSA/Experimental/Bits/Fast/Tactic.lean [hunk 2]
```diff
@@ -165,8 +222,7 @@ def test3 (x y : BitVec 300) : ((x ||| y) - (x ^^^ y)) = (x &&& y) := by
   bv_automata
 
 def test4 (x y : BitVec 2) : (x + -y) = (x - y) := by
-  bv_automata
-
+ bv_automata
```
Comment (tobiasgrosser) [inline SSA/Experimental/Bits/Fast/Tactic.lean:239]: Why do you drop the empty line between the two tests?

File: SSA/Experimental/Bits/Fast/Tactic.lean (hunk 1 above)
Comment (tobiasgrosser) [inline SSA/Experimental/Bits/Fast/Tactic.lean]:
```suggestion
  bv_automata
  
def test1 {w : Nat} (x y : BitVec (w + 1)) : (x ||| y) - (x ^^^ y) = x &&& y := by
```

File: SSA/Experimental/Bits/Fast/Tactic.lean [hunk 3]
```diff
@@ -103,21 +147,28 @@ Term.var 1 -- represent the 1st variable
...
-        q(ite ($lastBVar = $fVarId) $quotedCurrentFVar $accumulator)) (.fvar last)
-      let hyp : Q(Nat → BitStream) := q(fun _ => BitStream.ofBitVec $hypValue)
-      let newGoal : MVarId ← goal.define `vars hypType hyp
+      let length : Expr := a
+      let hypValue : Expr  := fVars.foldl (fun (accumulator : Expr) (currentFVar : FVarId) =>
+        let quotedCurrentFVar : Expr := .fvar currentFVar
+        let fVarId : Expr := quoteFVar currentFVar
+        iteE length lastBVar fVarId quotedCurrentFVar accumulator
+        ) (.fvar last)
+      let mapIndexToFVar : Expr := funE length hypValue --q(fun _ => BitStream.ofBitVec $hypValue)
```
Comment (tobiasgrosser) [inline SSA/Experimental/Bits/Fast/Tactic.lean]: Is this comment still needed?

File: SSA/Experimental/Bits/Fast/Tactic.lean [hunk 4]
```diff
@@ -199,3 +255,6 @@ def test24 (x y : BitVec 2) : (x ||| y) = (( x &&& (~~~y)) + y) := by
 
 def test25 (x y : BitVec 2) : (x &&& y) = (((~~~x) ||| y) - ~~~x) := by
   bv_automata
+
+def test26 {w : Nat} (x y : BitVec (w + 1)) : 1 + x + 0 = 1  + x := by
+  bv_automata
```
Comment (tobiasgrosser) [inline SSA/Experimental/Bits/Fast/Tactic.lean:275]: Maybe add a test case that has constants but is fixed-bitwidth and one that is arbitrary-bitwidth but does not use constants?

File: SSA/Experimental/Bits/Fast/Tactic.lean [hunk 5]
```diff
@@ -165,7 +236,7 @@ def test3 (x y : BitVec 300) : ((x ||| y) - (x ^^^ y)) = (x &&& y) := by
   bv_automata
 
 def test4 (x y : BitVec 2) : (x + -y) = (x - y) := by
-  bv_automata
+ bv_automata
```
Comment (alexkeizer) [inline SSA/Experimental/Bits/Fast/Tactic.lean:239]: You're still missing a single whitespace here

**Pull request merged**